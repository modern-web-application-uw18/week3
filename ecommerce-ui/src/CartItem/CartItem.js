import React, { Component } from 'react';
import PropTypes from 'prop-types';

class CartItem extends Component {
    render() {
        return(
            <div className="CartItem">
                <img 
                    src={this.props.item.image}
                    alt="house to rent"
                    className="CartItemImage"
                />
                <p className="FlexGrow1"></p>
                <p className="FlexGrow2">{this.props.item.title}</p>
                <p className="FlexGrow1"></p>
                <p className="FlexGrow1">Cost: ${this.props.item.payment.cost}</p>
                <button 
                    className="PurchaseButton" 
                    onClick={
                        this.props.onRemoveFromCart
                    }>
                    Remove from Cart
                </button>
            </div>
        );
    }
}

CartItem.propTypes = {
    item: PropTypes.object,
    onRemoveFromCart: PropTypes.func
}

export default CartItem;