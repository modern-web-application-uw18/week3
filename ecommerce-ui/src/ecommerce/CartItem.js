import React, { Component } from 'react';
import PropTypes from 'prop-types';

class CartItem extends Component {
  render() {
     return (
       <ul>
         <li>
             {this.props.title}{' ($' + this.props.cost + ') '}
             <button onClick={this.props.removeFromCart}
              className="cartBtn-r">Remove from cart</button>
         </li>
       </ul>
     );
   }
 }

CartItem.propTypes = {
    title: PropTypes.string,
    cost: PropTypes.number,
    removeFromCart: PropTypes.func
}

export default CartItem;
