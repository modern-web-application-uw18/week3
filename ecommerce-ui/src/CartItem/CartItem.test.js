import React from 'react';
import ReactDom from 'react-dom';
import TestRenderer from 'react-test-renderer';
import CartItem from './CartItem.js'

describe('CartItem', () => {
    it('should clear smoke test for CartItem rendering', () => {
        const div = document.createElement('div');
        const dummyFunc = (something) => {
            return () => {
                this.setState((prevState, props) => {
                    const purchasedUnit = something;
                    const newPurchasedUnits = prevState.purchasedUnits;
                    newPurchasedUnits.push(purchasedUnit);
                    return {
                        purchasedUnits: newPurchasedUnits
                    };
                });
            }
        };
        const dummyProp = {
            "title": "Cardboard Box",
            "houseType": "Apartment",
            "image": "./itsABox.jpg",
            "location": {
                "city": "Seattle",
                "country": "USA"
            },
            "payment": {
                "cost": 1,
                "description": "Free cancellation"
            },
            "host": {
                "name": "I5 underpass",
                "isSuperhost": false
            },
            "rating": {
                "stars": 1,
                "reviews": 6876
            }
        };
            
        ReactDom.render(<CartItem item={dummyProp} onRemoveFromCart={dummyFunc}/>, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it('should render a minimal CartItem section of Cart', () => {
        const dummyFunc = (index) => {
            return () => {
                this.setState((prevState, props) => {
                  const newPurchasedUnits = prevState.purchasedUnits;
                  newPurchasedUnits.splice(index, 1)
                  return {
                      purchasedUnits: newPurchasedUnits
                  }
                });
            }
        };
        const dummyProp = {
            "title": "Cardboard Box",
            "houseType": "Apartment",
            "image": "./itsABox.jpg",
            "location": {
                "city": "Seattle",
                "country": "USA"
            },
            "payment": {
                "cost": 1,
                "description": "Free cancellation"
            },
            "host": {
                "name": "I5 underpass",
                "isSuperhost": false
            },
            "rating": {
                "stars": 1,
                "reviews": 6876
            }
        };
            
        const component = TestRenderer.create(<CartItem item={dummyProp} onRemoveFromCart={dummyFunc}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
})