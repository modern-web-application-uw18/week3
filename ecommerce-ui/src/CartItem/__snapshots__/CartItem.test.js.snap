// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should render a minimal CartItem section of Cart 1`] = `
<div
  className="CartItem"
>
  <img
    alt="house to rent"
    className="CartItemImage"
    src="./itsABox.jpg"
  />
  <p
    className="FlexGrow1"
  />
  <p
    className="FlexGrow2"
  >
    Cardboard Box
  </p>
  <p
    className="FlexGrow1"
  />
  <p
    className="FlexGrow1"
  >
    Cost: $
    1
  </p>
  <button
    className="PurchaseButton"
    onClick={[Function]}
  >
    Remove from Cart
  </button>
</div>
`;
